src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Security;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+
+using SafeWinHttpHandle = Interop.WinHttp.SafeWinHttpHandle;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Static class containing the WinHttp global callback and associated routines.
+    /// </summary>
+    internal static class WinHttpRequestCallback
+    {
+        public static Interop.WinHttp.WINHTTP_STATUS_CALLBACK s_StaticCallbackDelegate =
```

Comment:
As notes in my PR description, there is a still a TOOD to reconcile the SafeHandle pattern. 

The patterns being used here (despite the code refactoring) has not changed from the current pattern used in WinHttpHandler.

As part of my future work on this, I will be writing tests to prove/disprove the GC theory and make sure the code is robust.